core/services/flux_monitor.go

```diff
@@ -247,18 +248,21 @@ type DeviationChecker interface {
 
 // PollingDeviationChecker polls external price adapters via HTTP to check for price swings.
 type PollingDeviationChecker struct {
-	initr        models.Initiator
-	address      common.Address
-	requestData  models.JSON
-	threshold    float64
-	precision    int32
-	runManager   RunManager
-	currentPrice decimal.Decimal
-	currentRound *big.Int
-	fetcher      Fetcher
-	delay        time.Duration
-	cancel       context.CancelFunc
-	newRounds    chan eth.Log
+	initr         models.Initiator
+	address       common.Address
+	requestData   models.JSON
+	idleThreshold time.Duration
+	threshold     float64
+	precision     int32
+	runManager    RunManager
+	currentPrice  decimal.Decimal
+	currentRound  *big.Int
+	fetcher       Fetcher
+	delay         time.Duration
+	cancel        context.CancelFunc
+	newRounds     chan eth.Log
+
+	stopC chan struct{}
```

Comment:
What's the significance of `C` in `stopC`? Is it indicating `Checker` or `Channel`?

Reply:
Its for channel, would you prefer another name or have any suggestions?

Reply:
I like the change to `waitOnStop`. Captures the intent well.